src/components/atoms/Button/Button.tsx

```diff
@@ -2,148 +2,83 @@ import React, { HTMLAttributes } from 'react';
-};
-
-const fontSizes: TButtonSizingMapping = {
-  compact: '14px',
-  default: '16px',
-  large: '20px',
-  small: '16px',
-};
-
-const paddings: TButtonSizingMapping = {
-  compact: '6px 16px',
-  default: '14px 32px',
-  large: '14px 32px',
-  small: '10px 24px',
+const colorMap = {
+  primary: {
+    text: colors.white,
+    bg: colors.action,
+    hover: `linear-gradient(90deg, #3B46C4 0%, #2732B0 100%)`,
+  },
+  secondary: {
+    text: colors.actionDark,
+    bg: colors.actionLight,
+    hover: shadeHexColor(colors.actionLight, 0.24),
+  },
+  link: {
+    text: colors.actionDark,
+    bg: 'transparent',
+    hover: shadeHexColor(colors.actionLight, 0.24),
+  },
 };
 
-const SText = styled.span<IButtonProps>`
-  display: block;
-  margin-right: ${({ rightIcon }) => !!rightIcon && '8px'};
-  margin-left: ${({ leftIcon }) => !!leftIcon && '8px'};
-`;
-
 export const buttonStyle = css<IButtonProps>`
   text-decoration: none;
   box-sizing: border-box;
   display: inline-flex;
   justify-content: center;
   align-items: center;
   width: ${({ fullWidth = false }) => fullWidth && '100%'};
-  padding: ${({ sizing = 'default' }) => paddings[sizing]};
+  padding: 15px 25px;
   font-family: ${typography.primary};
-  font-size: ${({ sizing = 'default' }) => fontSizes[sizing]};
+  font-size: ${typography.sizes.text};
   line-height: 1.2;
   font-weight: ${typography.weights.semibold};
   cursor: pointer;
-  border: 1px solid ${({ styling = 'primary' }) => backgroundColors[styling]};
   border-radius: 8px;
-  color: ${({ contrastColor }) => contrastColor};
-  color: ${({ styling = 'primary', contrastColor }) => (contrastColor ? contrastColor : fontColors[styling])};
-  background: ${({ styling = 'primary' }) => backgroundColors[styling]};
-  transition: all 140ms ease-in-out;
+  color: ${({ styling = 'primary' }) => colorMap[styling].text};
+  border: none;
 
-  &:not(:disabled) {
-    box-shadow: ${({ styling = 'primary' }) => boxShadows[styling] || 'none'};
-  }
+  ${({ styling = 'primary' }) => {
+    const { bg } = colorMap[styling];
+    const isActionGradient = bg === colors.action;
+    const bgFallback = `background-color: ${colors.actionPlain}`;
 
-  &:active:not(:disabled),
-  &:focus:not(:disabled) {
-    box-shadow: ${({ styling = 'primary' }) => activeBoxShadows[styling]};
-    ${({ styling }) => (styling === 'contrastLink' || styling === 'link' ? null : 'outline: none')};
-  }
+    return css`
+      ${isActionGradient && bgFallback}
+      background: ${bg};
+    `;
+  }}
 
   &:hover:not(:disabled) {
-    color: ${({ styling = 'primary' }) => hoverFontColors[styling]};
-    background-color: ${({ styling = 'primary' }) => activeBackgroundColors[styling]};
-    box-shadow: ${({ styling }) => (styling === 'contrastSecondary' ? activeBoxShadows[styling] : 'none')};
+    background: ${({ styling = 'primary' }) => colorMap[styling].hover};
   }
 
   &:active:not(:disabled) {
-    opacity: 0.4;
+    opacity: 0.8;
```

Comment:
[Opinion] what about adding `cursor: not-allowed;` ?

Reply:
This is when the button is `:active`, what would be the benefit to have `cursor: not-allowed` in this state?

It's set though for the `:disabled` state below.

Reply:
sorry! I meant for :disabled state!